starling/utils/QuadBatchVAOHelper.hx

```diff
@@ -0,0 +1,145 @@
+package starling.utils;
+
+#if gl_stats
+import openfl._internal.renderer.opengl.stats.DrawCallContext;
+import openfl._internal.renderer.opengl.stats.GLStats;
+#end
+
+import openfl._internal.stage3D.opengl.GLContext3D;
+import openfl._internal.stage3D.GLUtils;
+import flash.display3D.IndexBuffer3D;
+import starling.display.QuadBatch;
+import flash.display3D.Context3D;
+import lime.graphics.opengl.GLVertexArrayObject;
+
+
+@:access(openfl.display3D.Context3D)
+@:access(openfl.display3D.IndexBuffer3D)
+@:access(openfl.display3D.VertexBuffer3D)
+@:access(starling.display.QuadBatch)
+@:access(openfl._internal.stage3D.opengl.GLContext3D)
+@:access(openfl.display3D.Program3D)
+
+
+class QuadBatchVAOHelper {
+	
+	
+	public static function createVAO (context:Context3D):GLVertexArrayObject {
+		
+		#if vertex_array_object 
```

Comment:
Overall, you check "#if vertex_array_object" much more often in this PR compared to the last one. The last PR mainly just checks "vaoContext != null", would it make sense to make this more consistent either in one or the other direction?

Reply:
Here we use the compiler flag `vertex_array_object` to primarily to make starling working also with openfl versions that doesn't have implemented the VertexArrayObject improvement. So, what it does here, is cleaning up the dependencies towards the openfl's VertexArrayObject api and remove the redundant code.

Reply:
True point ;)